.github/workflows/build.yml

```diff
@@ -36,6 +36,37 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
+      # Check whether gradle.properties version has been released already
+      # This prevents accidently publish same version
+      # Only check binary for arithmetic as all published together and arithmetic is the first
+      - name: Check release published
+        run: |
+          # Skip if version contain SNAPSHOT
+          grep "^version" gradle.properties | grep -q SNAPSHOT || {
+            # Extract version from gradle.properties
+            VERSION=$(grep "^version" gradle.properties | sed 's|^version=\(.*\)$|\1|g')
+            echo "Query for version [$VERSION]"
+            RESPONSE_CODE=$(curl --head -L --silent --output /dev/null  --write-out %{http_code} "$ART_URL/$VERSION")
+            echo "DEBUG: Response code from curl command [$RESPONSE_CODE]"
+            if [[ $RESPONSE_CODE == "404" ]]
```

Comment:
so this is what we were hung up on previously?  a 404 would mean success, but we were failing because the curl exit code was non zero for a 404?

Reply:
Yes curl command returned non-zero because of 404 and I had not catch and returned zero return. It was a pure miss